src/app/prt-staking/components/prt-section/prt-widget.tsx

```diff
@@ -65,6 +65,7 @@ export function PrtWidget({ token, onClose }: Props) {
 
   const onClickTradeButton = useCallback(async () => {
     if (currentTab === WidgetTab.STAKE) {
+      // TODO: Should this happen before signing?
       if (!isApproved) {
         await onApprove()
       }
```

Comment:
@janndriessen think we should have the user approve the token prior to signing the stake message, or sign the stake message first and then approve?

Reply:
I'd leave this order to get get the longer task out of the way. Also, in case it something breaks.